### ISSUE TITLE
disable rule for files matching a patttern

### ISSUE THREAD

**Issue by kaddkaka** [PyCQA/pylint#10584]
### Question

We have a bunch of files using a specific filename pattern `".test.py$"`.

These have classes named with `snake_case`: 
```py
class test_my_banana_test: pass
```

Usually we want the `invalid-name` rule (`C0103`) to apply and make sure class names conforms to the pattern `"[A-Z_][a-zA-Z0-9]+$"`

Is it possible to disable this rule for just some files using one `pylintrc` or `pyproject.toml` file?

### Documentation for future user

I looked in here: 
1. https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html
1. the example pyproject.toml file: https://github.com/pylint-dev/pylint/blob/main/examples/pyproject.toml

**Comment by mbyrnepr2** [comment]
Hi!

> Is it possible to disable this rule for just some files using one pylintrc or pyproject.toml file?

Short answer: no - I was also a bit surprised by this.

There's a couple of related issues aleady:
https://github.com/pylint-dev/pylint/issues/2712
https://github.com/pylint-dev/pylint/issues/3767

In the latter issue there is a [comment](https://github.com/pylint-dev/pylint/issues/3767#issuecomment-1320019966) which announces a [package on PyPI ](https://pypi.org/project/pylint-per-file-ignores/)for this.

Other than that you possibly get away with using `good-names-rgxs = "test_"`; it applies to all files but maybe for your specific use-case it would be safe enough. Either way it's probably good add this feature (or document the above plugin if it looks appropriate).

**Comment by Pierre-Sassoulas** [comment]
We definitely need to use the same design than Ruff here (inspired by eslint if I remember correctly). Possibly even take Ruff pylint file disable into account as if they were our own disable when we don't have anything in the pylint file disable.
